Add materializecss gem and files.

ymnj opened the pull request:
Create controllers for sessions and users.
Create view forms for sessions and users using materializecss.
Create routes for sessions and users
Add initial login and sign up links to application view.
Add Simple_form
Add create ability to sign up users through users#create

ymnj committed:
Add materializecss gem and files.
Create controllers for sessions and users.
Create view forms for sessions and users using materializecss.
Create routes for sessions and users
Add initial login and sign up links to application view.
Add Simple_form
Add create ability to sign up users through users#create

zibs commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,26 @@
+      redirect_to "/"
+    else 
+      flash[:notice] = "Please fill in the required fields"
+      render '/signup'
```
We shouldn't hardcode paths, but instead use the Rails built in path helper like 

``` ruby
redirect_to user_path(@user)
...
else
render :new
```

samweatherly commented on app/controllers/sessions_controller.rb:
```diff
@@ -0,0 +1,25 @@
+      session[:user_id] = user.id
+      redirect_to root_path, notice: "Thank you for logging in"
+    else
+      flash[:notice] = "Username or password invalid"
```
flash[:alert] may be better suited here.

ymnj committed:
Add user methods to application controller
User sessions are created and stored on log in.
Add login and logout links to application view. Show or hide based on if user signed in.
